kbcstorage/retry_requests.py

```diff
@@ -0,0 +1,31 @@
+import time
+import requests
+
+MAX_RETRIES = 5
+BACKOFF_FACTOR = 1.0
+RETRY_STATUS_CODES = {500, 502, 503, 504}
+
+
+def _get_backoff_time(retry_count):
```

Comment:
I think it would be better to have the retry faster by default. 

For factor 0.25
![image](https://github.com/user-attachments/assets/98ba7b11-d157-4e92-85ee-68a5db0ba23b)

It still means that the request will take 15s in the worst case scenario.

Reply:
Although I can see that we have the same in PHP client :/ 

https://github.com/keboola/storage-api-php-client/blob/37889cd2922f0094cd9b34378b5c95dea9aa5eea/src/Keboola/StorageApi/HandlerStack.php#L85C5-L90C6

Reply:
BTW: Ideally the logic should be the same https://github.com/keboola/storage-api-php-client/blob/37889cd2922f0094cd9b34378b5c95dea9aa5eea/src/Keboola/StorageApi/HandlerStack.php#L53-L83, for example it retries anything > 499, not just some selected 50x errors

Reply:
Yes, the behavior should be the same, incl. the timeouts

Reply:
Take a look again please, I adjusted the logic, it should now be the same as it is in php implementation.